From 199a2d247b1255af78d988a6c8beb6674d6d6b91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Fri, 5 May 2023 11:05:26 +0000
Subject: [PATCH 2/3] Reapply [test] [llvm-config] Assume unix style lib names
 on mingw targets

Assume the MSVC style naming only for "windows-msvc" targets.

Differential Revision: https://reviews.llvm.org/D149998
---
 llvm/test/lit.cfg.py                                 | 4 ++++
 llvm/test/tools/llvm-config/system-libs.test         | 2 +-
 llvm/test/tools/llvm-config/system-libs.windows.test | 2 +-
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/llvm/test/lit.cfg.py b/llvm/test/lit.cfg.py
index 72f16adddebc..59bfd8ecfd13 100644
--- a/llvm/test/lit.cfg.py
+++ b/llvm/test/lit.cfg.py
@@ -365,6 +365,10 @@ if config.target_triple:
     if not config.target_triple.startswith(("nvptx", "xcore")):
         config.available_features.add('object-emission')
 
+# Allow checking for specific details in the host triple
+if config.host_triple:
+    config.available_features.add('host=%s' % config.host_triple)
+
 if config.have_llvm_driver:
   config.available_features.add('llvm-driver')
 
diff --git a/llvm/test/tools/llvm-config/system-libs.test b/llvm/test/tools/llvm-config/system-libs.test
index 708ce3ef99a4..e6450a6f5a0e 100644
--- a/llvm/test/tools/llvm-config/system-libs.test
+++ b/llvm/test/tools/llvm-config/system-libs.test
@@ -1,6 +1,6 @@
 RUN: llvm-config --link-static --system-libs Support 2>&1 | FileCheck %s
 REQUIRES: static-libs
-UNSUPPORTED: system-windows
+UNSUPPORTED: host={{.*-windows-msvc}}
 CHECK: -l
 CHECK-NOT: error
 CHECK-NOT: warning
diff --git a/llvm/test/tools/llvm-config/system-libs.windows.test b/llvm/test/tools/llvm-config/system-libs.windows.test
index 55d1fbea9788..d1d91a9a0b9e 100644
--- a/llvm/test/tools/llvm-config/system-libs.windows.test
+++ b/llvm/test/tools/llvm-config/system-libs.windows.test
@@ -1,6 +1,6 @@
 RUN: llvm-config --link-static --system-libs Support 2>&1 | FileCheck %s
 REQUIRES: static-libs
-REQUIRES: system-windows
+REQUIRES: host={{.*-windows-msvc}}
 CHECK-NOT: -l
 CHECK: psapi.lib shell32.lib ole32.lib uuid.lib advapi32.lib
 CHECK-NOT: error
-- 
2.34.1

